### PR TITLE
Merge new model structuring

### PR DESCRIPTION
**Pull request by davidwyld**
Now this is decided on and tested, we should be able to merge @fredblain's tidier model class structure.

**Note:** the push CI tests will fail because the `model_zoo` branch still has the old nmt-keras travis config.

### CODE REVIEW

**Comment by davidwyld** [comment]
So it looks like with the new model structure, we get slightly different results for document level as well as sentence level in comparison to the old model structure. Word level gets the same results. 
In some senses this is unsurprising, since BiRNN doc level uses the same model as sentence level (`EncSent`).

In summary, the best pearson correlation coefficients (with the seed fixed) are:

**Sentence level**

|  Backend  |  New Models  |  Old Models  |
| :----------- | :---------------- | :-------------- |
|  Theano  |  0.17946171830618374  |  0.1910756629756321  |
|  TensorFlow  |  0.18309480272019577  |  0.16354782921305733  |

**Document level**

|  Backend  |  New Models  |  Old Models  |
| :----------- | :---------------- | :-------------- |
|  Theano  |  0.7243370113216334  |  0.7583320076869026  |
|  TensorFlow  |  0.7323550350999823  |  0.6443940541110599  |



(will update these when I run them)
**Word level**

|  Backend  |  New Models  |  Old Models  |
| :----------- | :---------------- | :-------------- |
|  Theano  |    |    |
|  TensorFlow  |    |    |
